src/libraries/Common/src/System/Reflection/Metadata/TypeNameParserHelpers.cs

```diff
@@ -0,0 +1,377 @@
+            }
+            else
+            {
+                Debug.Assert(rankOrModifier >= 2);
+
+                builder.Append('[');
+                builder.Append(',', rankOrModifier - 1);
+                builder.Append(']');
+            }
+
+            return builder.ToString();
+        }
+
+        /// <summary>
+        /// Are there any captured generic args? We'll look for "[[" and "[" that is not followed by "]", "*" and ",".
+        /// </summary>
+        internal static bool IsBeginningOfGenericArgs(ref ReadOnlySpan<char> span, out bool doubleBrackets)
+        {
+            doubleBrackets = false;
+
+            if (!span.IsEmpty && span[0] == '[')
+            {
+                // There are no spaces allowed before the first '[', but spaces are allowed after that.
+                ReadOnlySpan<char> trimmed = span.Slice(1).TrimStart();
+                if (!trimmed.IsEmpty)
+                {
+                    if (trimmed[0] == '[')
+                    {
+                        doubleBrackets = true;
+                        span = trimmed.Slice(1).TrimStart();
+                        return true;
+                    }
+                    if (!(trimmed[0] is ',' or '*' or ']')) // [] or [*] or [,] or [,,,, ...]
+                    {
+                        span = trimmed;
+                        return true;
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        internal static bool TryGetTypeNameInfo(ref ReadOnlySpan<char> input, ref List<int>? nestedNameLengths, out int totalLength)
+        {
+            bool isNestedType;
+            totalLength = 0;
+            do
+            {
+                int length = GetFullTypeNameLength(input.Slice(totalLength), out isNestedType);
+                if (length <= 0)
+                {
+                    // invalid type names:
+                    // -1: invalid escaping
+                    // 0: pair of unescaped "++" characters
+                    return false;
+                }
+
+#if SYSTEM_PRIVATE_CORELIB
+                // Compat: Ignore leading '.' for type names without namespace. .NET Framework historically ignored leading '.' here. It is likely
+                // that code out there depends on this behavior. For example, type names formed by concatenating namespace and name, without checking for
+                // empty namespace (bug), are going to have superfluous leading '.'.
+                // This behavior means that types that start with '.' are not round-trippable via type name.
+                if (length > 1 && input[0] == '.' && input.Slice(0, length).LastIndexOf('.') == 0)
+                {
+                    input = input.Slice(1);
+                    length--;
+                }
+#endif
+                if (isNestedType)
+                {
+                    // do not validate the type name now, it will be validated as a whole nested type name later
```

Comment:
What kind of validation is this comment talking about?

Reply:
The one that I've removed a while ago. I am going to remove the outdated comment ;)